src/vivarium/framework/population.py

```diff
@@ -237,8 +238,16 @@ def _get_view(self, columns: Sequence[str], query: str = None):
         return PopulationView(self, columns, query)
 
     def register_simulant_initializer(self, initializer: Callable,
-                                      creates_columns: Sequence[str] = (), requires_columns: Sequence[str] = ()):
-        self.register_population_initializer((initializer, creates_columns, tuple(requires_columns) + ('tracked',)))
+                                      creates_columns: List[str] = (),
+                                      requires_columns: List[str] = (),
+                                      requires_values: List[str] = (),
+                                      requires_streams: List[str] = ()):
+        dependencies = ([f'column.{name}' for name in requires_columns]
+                        + [f'value.{name}' for name in requires_values]
+                        + [f'stream.{name}' for name in requires_streams])
+        if creates_columns != ['tracked']:
```

Comment:
New:  The tracked column is a dependency of everything since it's used by the population view directly.